ip: allow extra IP address found in verification stage

cathay4t opened the pull request:
When nmstate applying the network state, there might be another tool
changing the IP addresses of interface, which lead to verification error
as extra IP address found. This is valid use case in kubernetes-nmstate
where keepalived is trying to add VIP(192.168.111.4) to certain interface.

To support that, we introduce `InterfaceIpv4.allow_extra_ip` and
`InterfaceIpv6.allow_extra_ip` with default set to true, nmstate
verification will ignore extra IP address after applied.

Considering this is a very corner case, and could make the life of of
OpenshiftSDN engineer easier, I would suggest we accept this breaker of
API behavior.

It is hard to reproduce it in integration test case, hence only added
unit test cases.

ffmancera reviewed (changes requested):
Integration tests are failing, please take look. Thanks!

cathay4t committed:
ip: allow extra IP address found in verification stage

When nmstate applying the network state, there might be another tool
changing the IP addresses of interface, which lead to verification error
as extra IP address found. This is valid use case in kubernetes-nmstate
where keepalived is trying to add VIP(192.168.111.4) to certain interface.

To support that, we introduce `InterfaceIpv4.allow_extra_ip` and
`InterfaceIpv6.allow_extra_ip` with default set to true, nmstate
verification will ignore extra IP address after applied.

Considering this is a very corner case, and could make the life of of
OpenshiftSDN engineer easier, I would suggest we accept this breaker of
API behavior.

It is hard to reproduce it in integration test case, hence only added
unit test cases.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
CI failure fixed. This is caused by `des_ip_addrs.contains(i)` is comparing mptcp flags where should be ignored when dropping extra IP. Patch updated.
